src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,164 @@
+                        }
+                    }
+                }
+            }
+        } else if (input.startsWith(this.event + this.space)) {
+            String s = input.substring((this.event + this.space).length());
+            if (!s.contains(space + eventAt + space)) {
+                if (s.startsWith(eventAt)) {
+                    System.out.println("The event is empty, what do you really mean?");
+                } else {
+                    System.out.println("The time is empty, do you mean it never starts?");
+                }
+            } else {
+                String event = s.substring(0, s.indexOf(eventAt) - space.length());
+                String time = s.substring(s.indexOf(eventAt) + eventAt.length()
+                        + space.length());
+                if (event.isBlank()) {
+                    System.out.println("The event is empty, what do you really mean?");
+                } else if (time.isBlank()) {
+                    System.out.println("The time is empty, do you mean it never starts?");
+                } else {
+                    LocalDate t = null;
+                    try {
+                        t = LocalDate.parse(time);
+                        d.addEvent(event, t);
+                        try {
+                            d.getStorage().writeToFile(this.event + this.space + s + "\n", isLoading);
+                        } catch (IOException e) {
+                            System.err.println("This command could not be saved due to an unknown error.");
+                        }
+                    } catch (DateTimeParseException e) {
+                        d.addEvent(event, time);
+                        try {
+                            d.getStorage().writeToFile(this.event + this.space + s + "\n", isLoading);
+                        } catch (IOException ex) {
+                            System.err.println("This command could not be saved due to an unknown error.");
+                        }
+                    }
+                }
+            }
+        } else if (input.startsWith(this.delete + this.space)) {
+            try {
+                String parameter = input.substring((this.delete + this.space).length());
+                int param = Integer.parseInt(parameter);
+                d.deleteTask(param);
+                try {
+                    d.getStorage().writeToFile(this.delete + this.space + parameter + "\n", isLoading);
+                } catch (IOException e) {
+                    System.err.println("This command could not be saved due to an unknown error.");
+                }
+            } catch (NumberFormatException e) {
+                System.err.println("Invalid task.");
+            }
+        } else if (input.startsWith(this.find + this.space)) {
+            String s = input.substring((this.find + this.space).length());
+            if (s.isBlank()) {
```

Comment:
I think you can avoid deep nesting by using switch.

Reply:
Switch statements with the current code structure would be too complicated.
Instead, I will shift the bulk of the processes to private methods of the parser class and have the parser call those methods instead. Apparently, the code was not following the single level of abstraction principle which resulted in a very long parse method.